t/001-tls.t

```diff
@@ -521,3 +521,206 @@ nil, connection is not TLS or TLS support for Nginx not enabled
+            end
+            ngx.sleep(2)
+            assert(handle:write("bad request\n"))
+            handle:close()
+
+            handle = io.popen("grep '^Acceptable client certificate CA names$\\|^C = US,' /tmp/output.txt")
+            if not handle then
+                ngx.log(ngx.ERR, "unable to popen grep: ", err)
+                return ngx.exit(ngx.ERROR)
+            end
+            ngx.print(handle:read("*a"))
+            handle:close()
+        }
+    }
+
+--- request
+GET /t
+--- response_body
+Acceptable client certificate CA names
+C = US, ST = California, O = Kong Testing, CN = Kong Testing Intermidiate CA
+C = US, ST = California, O = Kong Testing, CN = Kong Testing Root CA
+
+--- error_log
+ssl cert by lua is running!
+ssl cert by lua complete!
+
+--- no_error_log
+[error]
+[alert]
+[warn]
+[crit]
+
+
+
+=== TEST 8: without calling set_client_ca_list, ca dn list isn't sent
+--- http_config
+    lua_package_path "../lua-resty-core/lib/?.lua;lualib/?.lua;;";
+
+    server {
+        listen unix:$TEST_NGINX_HTML_DIR/nginx.sock ssl;
+        server_name   example.com;
+        ssl_certificate_by_lua_block {
+            print("ssl cert by lua is running!")
+
+            local tls_lib = require "resty.kong.tls"
+            local suc, err
+
+            suc, err = tls_lib.request_client_certificate()
+            if err then
+                ngx.log(ngx.ERR, "unable to request client certificate: ", err)
+                return ngx.exit(ngx.ERROR)
+            end
+
+            print("ssl cert by lua complete!")
+        }
+        ssl_certificate ../../cert/example.com.crt;
+        ssl_certificate_key ../../cert/example.com.key;
+        ssl_session_tickets off;
+
+        server_tokens off;
+        location /foo {
+            default_type 'text/plain';
+            content_by_lua_block {
+                ngx.say("impossibe to reach here")
+            }
+            more_clear_headers Date;
+        }
+    }
+--- config
+    server_tokens off;
+
+    location /t {
+        content_by_lua_block {
+            local handle = io.popen("openssl s_client -unix $TEST_NGINX_HTML_DIR/nginx.sock > /tmp/output.txt", "w")
+            if not handle then
+                ngx.log(ngx.ERR, "unable to popen openssl: ", err)
+                return ngx.exit(ngx.ERROR)
+            end
+            ngx.sleep(2)
+            assert(handle:write("bad request\n"))
+            handle:close()
+
+            handle = io.popen("grep '^No client certificate CA names sent$' /tmp/output.txt")
+            if not handle then
+                ngx.log(ngx.ERR, "unable to popen grep: ", err)
+                return ngx.exit(ngx.ERROR)
+            end
+            ngx.print(handle:read("*a"))
+            handle:close()
+        }
+    }
+
+--- request
+GET /t
+--- response_body
+No client certificate CA names sent
+
+--- error_log
+ssl cert by lua is running!
+ssl cert by lua complete!
+
+--- no_error_log
+[error]
+[alert]
+[warn]
+[crit]
```

Comment:
add test for `ssl.parse_pem_cert` function to get `STACK_OF(X509) *`?

Reply:
I think it should be a test of `ngx.ssl` itself. We only focus on the APIs of our own module.

Reply:
i agree having such test will be good, as `ssl.parse_pem_cert` doesn't guaranteed to always work with our library, it's designed to be consumed
by another upstream API, so it could change internally without our notice.

Reply:
Okay, I will add the test.